tests/suites/test_suite_psa_crypto_util.function

```diff
@@ -0,0 +1,74 @@
+/* BEGIN_HEADER */
+#include <test/helpers.h>
+#include <mbedtls/psa_util.h>
+/* END_HEADER */
+
+/* BEGIN_CASE depends_on:MBEDTLS_PSA_UTIL_HAVE_ECDSA */
+void ecdsa_raw_to_der(int key_bits, data_t *input, data_t *exp_result, int exp_ret)
+{
+    unsigned char *tmp_buf = NULL;
+    size_t tmp_buf_len = exp_result->len;
+    size_t ret_len;
+
+    TEST_CALLOC(tmp_buf, tmp_buf_len);
+
+    TEST_EQUAL(mbedtls_ecdsa_raw_to_der(input->x, input->len,
+                                        tmp_buf, tmp_buf_len, &ret_len,
+                                        key_bits), exp_ret);
+
+    if (exp_ret == 0) {
+        ASSERT_COMPARE(exp_result->x, exp_result->len, tmp_buf, ret_len);
+    }
+
+exit:
+    mbedtls_free(tmp_buf);
+}
+/* END_CASE */
+
+/* BEGIN_CASE depends_on:MBEDTLS_PSA_UTIL_HAVE_ECDSA */
+void ecdsa_raw_to_der_incremental(int key_bits, data_t *input, data_t *exp_result)
+{
+    unsigned char *tmp_buf = NULL;
+    size_t tmp_buf_len = exp_result->len;
+    size_t ret_len;
+    size_t i;
+
+    for (i = 1; i < tmp_buf_len; i++) {
+        TEST_CALLOC(tmp_buf, i);
+        TEST_ASSERT(mbedtls_ecdsa_raw_to_der(input->x, input->len,
+                                             tmp_buf, i, &ret_len,
+                                             key_bits) != 0);
+        mbedtls_free(tmp_buf);
+        tmp_buf = NULL;
+    }
+
+    TEST_CALLOC(tmp_buf, i);
+    TEST_EQUAL(mbedtls_ecdsa_raw_to_der(input->x, input->len,
+                                        tmp_buf, i, &ret_len,
+                                        key_bits), 0);
+exit:
+    mbedtls_free(tmp_buf);
+}
+/* END_CASE */
+
+/* BEGIN_CASE depends_on:MBEDTLS_PSA_UTIL_HAVE_ECDSA */
+void ecdsa_der_to_raw(int key_bits, data_t *input, data_t *exp_result, int exp_ret)
+{
+    unsigned char *tmp_buf = NULL;
+    size_t tmp_buf_len = exp_result->len;
+    size_t ret_len;
+
+    TEST_CALLOC(tmp_buf, tmp_buf_len);
+
```

Comment:
We should have more testing of invalid data that has a risk of causing a buffer overflow. A good way would be to test all truncations of the input and check that they cause an error, like `parse_prefixes` in `test_suite_asn1parse`.